compiler/parser/src/parser/type_.rs

```diff
@@ -108,7 +108,7 @@ impl<N: Network> ParserContext<'_, N> {
                 }
             }
 
-            Ok((Type::Composite(CompositeType { id: ident, program: self.program_name }), ident.span))
+            Ok((Type::Composite(CompositeType { id: ident, program: None }), ident.span))
```

Comment:
Just curious, do you see a path to resolve composite type names so that we always know the program after some pass? I'm not sure it gives us anything yet, but it feels like a good invariant to have.

Reply:
I think it's sufficient.